module/zfs/vdev.c

```diff
@@ -4598,7 +4598,14 @@ vdev_deadman(vdev_t *vd, char *tag)
 void
 vdev_set_deferred_resilver(spa_t *spa, vdev_t *vd)
 {
-	ASSERT(vd->vdev_ops->vdev_op_leaf);
+	for (uint64_t i = 0; i < vd->vdev_children; i++) {
+		vdev_set_deferred_resilver(spa, vd->vdev_child[i]);
+	}
+
+	if (!vd->vdev_ops->vdev_op_leaf || !vdev_writeable(vd) ||
+	    range_tree_is_empty(vd->vdev_dtl[DTL_MISSING]))
+		return;
```

Comment:
nit: It's a little thing but I'd suggest using `{ }` here, and removing them from the above `for()` loop.